File: app/src/main/java/com/mohammedsazid/android/aiub/CustomWebView.java
package com.mohammedsazid.android.aiub;

import android.content.Context;
import android.util.AttributeSet;

import java.io.File;

public class CustomWebView extends AdvancedWebView {
    public CustomWebView(Context context) {
        super(context);
        setupWebView();
    }

    public CustomWebView(Context context, AttributeSet attrs) {
        super(context, attrs);
        setupWebView();
    }

    public CustomWebView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        setupWebView();
    }

    private void setupWebView() {
        setCookiesEnabled(true);
        setThirdPartyCookiesEnabled(true);
        setDesktopMode(false);
        getSettings().setSupportZoom(true);
        getSettings().setBuiltInZoomControls(true);
        getSettings().setDisplayZoomControls(false);
        getSettings().setLoadsImagesAutomatically(true);

        addHttpHeader("X-Requested-With", getContext().getString(R.string.app_name));
        getSettings().setAppCacheEnabled(true);
        getSettings().setAppCachePath(getContext().getCacheDir().getAbsolutePath()
                + File.separator + "appCache" + File.separator);
        setSaveEnabled(true);
    }
}

File: app/src/main/java/com/mohammedsazid/android/aiub/widgets/CustomWebView.kt
package com.mohammedsazid.android.aiub.widgets

import android.content.Context
import android.util.AttributeSet

import com.mohammedsazid.android.aiub.R

import java.io.File

class CustomWebView : AdvancedWebView {
    constructor(context: Context) : super(context) {
        setupWebView()
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        setupWebView()
    }

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        setupWebView()
    }

    private fun setupWebView() {
        setCookiesEnabled(true)
        setThirdPartyCookiesEnabled(true)
        setDesktopMode(false)
        settings.setSupportZoom(true)
        settings.builtInZoomControls = true
        settings.displayZoomControls = false
        settings.loadsImagesAutomatically = true

        addHttpHeader("X-Requested-With", context.getString(R.string.app_name))
        settings.setAppCacheEnabled(true)
        settings.setAppCachePath(context.cacheDir.absolutePath
                + File.separator + "appCache" + File.separator)
        isSaveEnabled = true
    }
}
